packages/vscode-extension/src/commands/ListFiles.ts

```diff
@@ -0,0 +1,86 @@
+/*
+ ** Copyright (c) 2021 Oracle and/or its affiliates.  All rights reserved.
+ ** Licensed under the Universal Permissive License v 1.0 as shown at https://oss.oracle.com/licenses/upl.
+ */
+import { window, QuickPickItem } from 'vscode';
+import { COMMAND, LIST_FILES } from '../service/TranslationKeys';
+import { actionResultStatus, AccountFileCabinetService, ExecutionEnvironmentContext } from '../util/ExtensionUtil';
+
+import BaseAction from './BaseAction';
+import { FolderType } from './FolderType';
+import { getSdkPath } from '../core/sdksetup/SdkProperties';
+import * as vscode from 'vscode';
+import { VSCODE_PLATFORM } from '../ApplicationConstants';
+
+const COMMAND_NAME = 'listfiles';
+
+const LIST_FILES_COMMAND = {
+	OPTIONS: {
+		FOLDER: 'folder',
+	},
+};
+
+export default class ListFiles extends BaseAction {
+	constructor() {
+		super(COMMAND_NAME);
+	}
+
+	protected async execute(): Promise<void> {
+		const fileCabinetFolders = await this._getListFolders();
+		const selectedFolder = await this._selectFolder(fileCabinetFolders);
+
+		if (selectedFolder === null || selectedFolder === undefined) {
+			return;
+		}
+
+		await this._listFiles(selectedFolder.label);
+	}
+
+	protected async _getListFolders() {
+		const executionEnvironmentContext = new ExecutionEnvironmentContext({
+			platform: VSCODE_PLATFORM,
+			platformVersion: vscode.version,
+		});
+
+		const listFoldersPromise = AccountFileCabinetService.getFileCabinetFolders(
+			getSdkPath(),
+			executionEnvironmentContext,
+			this.executionPath,
+			COMMAND_NAME
+		);
+		const statusBarMessage = this.translationService.getMessage(LIST_FILES.LOADING_FOLDERS);
+		this.messageService.showStatusBarMessage(statusBarMessage, listFoldersPromise);
+
+		return listFoldersPromise;
+	}
+
+	protected async _selectFolder(filteredFolders: {
+		map: (arg0: (folder: FolderType) => QuickPickItem) => QuickPickItem[];
```

Comment:
This part looks pretty weird to me. What do you accomplish with it?

Reply:
I refer to `filteredFolders: { map: (arg0: (folder: FolderType) => QuickPickItem) => QuickPickItem[]; }`
We have decided not to hide the restricted folders, then I think `filteredFolders` doesn't sound good.

Reply:
Ok to the change of the name.
I need to specify QuickPickItem here because if not it would infer a string.
I need a QuickPickItem to set the "Restricted Folder" in the detail (it would appeear with special typing font)

Reply:
Sorry, when I said "detail" I wanted to say "description", "detail" is another property inside QuickPickItem that is slightly different to "description"